pdata/internal/wrapper_traces.go

```diff
@@ -20,15 +20,34 @@ import (
 )
 
 type Traces struct {
-	orig *otlpcollectortrace.ExportTraceServiceRequest
+	orig  *otlpcollectortrace.ExportTraceServiceRequest
+	state *State
+}
+
+func (ms *Traces) EnsureMutability() {
+	if *ms.state == StateShared {
+		*ms.state = StateDirty
+		newResourceSpans := clone(ms.orig.ResourceSpans) // clone to be implemented for otlp only
```

Comment:
newOrig := ... 
newResourceSpan(ms.orig.ResourceSpans).CopyTo(newResourceSpan(newOrig))

Reply:
`newResourceSpan` needs pointers to a state and parent now

Reply:
Updated